### PR TITLE
#160370638 Delete comment functionality

### PR DESCRIPTION
**Pull request by ikenjoku**
#### What does this PR do?
This PR implements the delete comment functionality which enables users to delete their own comments posted on a request.

#### Description of Task to be completed?
- Write tests for delete component
- Create modal to confirm comment delete
- Fix failing tests

#### How should this be manually tested?
- Clone the repository
- Run `make start` docker build command on the terminal to install all dependencies and run the App in the docker development environment server.
- On the browser, type the localhost URL ` http://localhost:3000` and Login with your Andela email account into the App.
- On the Requests page, click on the request Id from the list of requests displayed on the request table, you should be able to see the Request details Modal displayed.
- Add a new comment using the comment box.
- On the displayed added comment, click on the `delete` button to delete the comment.
- When done, you should be able to see that the deleted comment is gone.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#160370638](https://www.pivotaltracker.com/story/show/160370638)

#### Screenshots (if appropriate)
screen shot 2018-09-26 at 12 41 21 pm

screen shot 2018-09-26 at 12 41 09 pm

#### Questions:
None

### CODE REVIEW

**Comment by matthewacha** [comment]
@ikenjoku  I would like to know why you chose to position the popup asymmetrically? I was wondering if you could try to centre it so that  It blocks the direct line of sight to all other content behind it and captures the attention of the user. 
For the delete message, please refer to this article https://uxplanet.org/how-to-write-good-error-messages-858e4551cd4

**Comment by ikenjoku** [comment]
Thanks for the review @matthewacha , the link was great. The popup is positioned to reproduce the design specs. This is the mock below
https://projects.invisionapp.com/d/main#/console/14861108/320832646/preview

**Comment by ikenjoku** [comment]
Thank you @Nedson202 , I will squash the commits. Helpful link.

File: src/components/modal/Modal.jsx [hunk 1]
```diff
@@ -55,15 +70,19 @@ class Modal extends PureComponent {
...
+      customModalStyles
     } = this.props;
 
+    const overlayStyle = `${visibility} ${customOverlayStyle}`;
+
     return (
       visibility === 'visible' ? (
         <Fragment>
-          <Overlay click={closeModal} className={visibility}>
+          <Overlay className={overlayStyle} overlayBg={!showOverlay ? 'overlayBg' : ''}>
```
[the remark this answers is not in the document]
Comment (ikenjoku) [reply src/components/modal/Modal.jsx]: Thanks @Nedson202, it is done.

**Comment by ikenjoku** [comment]
That's right @matthewacha . I've updated the delete link and the modal position. I think the current message is brief and clear.

**Comment by babadee001** [comment]
@ikenjoku Please fix merge conflicts and drop in coverage

**Pull request merged**